src/groupeddataframe/grouping.jl

```diff
@@ -115,71 +118,102 @@ Base.getindex(gd::GroupedDataFrame, I::AbstractArray{Bool}) =
 Base.names(gd::GroupedDataFrame) = names(gd.parent)
 _names(gd::GroupedDataFrame) = _names(gd.parent)
 
-##############################################################################
-##
-## GroupApplied...
-##    the result of a split-apply operation
-##    TODOs:
-##      - better name?
-##      - ref
-##      - keys, vals
-##      - length
-##      - start, next, done -- should this return (k,v) or just v?
-##      - make it a real associative type? Is there a need to look up key columns?
-##
-##############################################################################
-
-"""
-The result of a `map` operation on a GroupedDataFrame; mainly for use
-with `combine`
-
-Not meant to be constructed directly, see `groupby` abnd
-`combine`. Minimal support is provided for this type. `map` is
-provided for a GroupApplied object.
-
-"""
-struct GroupApplied{T<:AbstractDataFrame}
-    gd::GroupedDataFrame
-    vals::Vector{T}
-
-    function (::Type{GroupApplied})(gd::GroupedDataFrame, vals::Vector)
-        length(gd) == length(vals) ||
-            throw(DimensionMismatch("GroupApplied requires keys and vals be of equal length (got $(length(gd)) and $(length(vals)))."))
-        new{eltype(vals)}(gd, vals)
-    end
-end
-
-
-#
-# Apply / map
-#
-
-# map() sweeps along groups
-function Base.map(f::Function, gd::GroupedDataFrame)
-    GroupApplied(gd, [wrap(f(df)) for df in gd])
-end
-function Base.map(f::Function, ga::GroupApplied)
-    GroupApplied(ga.gd, [wrap(f(df)) for df in ga.vals])
-end
-
 wrap(df::AbstractDataFrame) = df
+wrap(nt::NamedTuple) = nt
 wrap(A::Matrix) = convert(DataFrame, A)
-wrap(s::Any) = DataFrame(x1 = s)
+wrap(s::Union{AbstractVector, Tuple}) = DataFrame(x1 = s)
```

Comment:
what is the idea of wrapping a tuple like this?

As the code gives us:
```
julia> DataFrame(x=(1,2,3))
1×1 DataFrame
│ Row │ x         │
│     │ Tuple…    │
├─────┼───────────┤
│ 1   │ (1, 2, 3) │
```
(why do we prefer `DataFrame` to `NamedTuple` in this case?)

Reply:
Hmm. I think I incorrectly assumed that the existing code treated tuples like vectors, but clearly it doesn't. I'll change that to use `NamedTuple` as it's more efficient.